Add Chromium v65 Playready support

petegriffin opened the pull request:
With these patches applied Playready content is being played with the following command

chromium --no-sandbox --start-maximized --user-data-dir=data_dir --enable-logging --v=0 --unsafely-treat-insecure-origin-as-secure="http://people.linaro.org" --register-pepper-plugins="/usr/lib64/chromium/libopencdmadapter.so#ClearKey CDM#ClearKey CDM0.1.0.0#1.0.0;application/x-ppapi-open-cdm" http://people.linaro.org/~peter.griffin/dash.js/samples/dash-if-reference-player/index.html

petegriffin committed:
BUILD.gn: Update to use build_flag_header.gni.

This updates to use auto generated opencdm_features.h header.
Also add the cdm_logging files to avoid linker errors when
doing debug builds.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
opencdm_key_systems.cc: Update to use USE_PROPRIETARY_CODECS buildflag

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
ocdm_encrypted_media_istypesupported_browsertest: Update to use USE_PROPRIETARY_CODECS buildflag

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
BUILD.gn: add opencdm_features to dependencies

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
opencdm_key_systems.cc: add proper path to generated header

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin committed:
Update OCDM_USE_PLAYREADY to use BUILDFLAG

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

ajutras-linaro commented on src/browser/opencdm_key_systems.cc:
```diff
@@ -20,6 +20,7 @@
 #include "media/base/key_system_properties.h"
 #include "media/media_features.h"
 #include "opencdm_key_systems.h"
+#include "media/cdm/ppapi/external_open_cdm/src/browser/opencdm_features.h"
```
This will only work on Chromium. Can you just include "opencdm_features.h"?

ajutras-linaro commented on src/common/open_cdm_common.h:
```diff
@@ -17,7 +17,9 @@
 #define MEDIA_CDM_PPAPI_EXTERNAL_OPEN_CDM_COMMON_OPEN_CDM_COMMON_H_
 
-#ifdef OCDM_USE_PLAYREADY
+#include "media/cdm/ppapi/external_open_cdm/src/browser/opencdm_features.h"
```
Chromium specific. Can you directly include "opencdm_features.h"?

ajutras-linaro commented on src/mediaengine/open_cdm_mediaengine_factory.h:
```diff
@@ -21,12 +21,12 @@
 #include "media/cdm/ppapi/external_open_cdm/src/mediaengine/open_cdm_mediaengine_impl.h"
-
+#include "media/cdm/ppapi/external_open_cdm/src/browser/opencdm_features.h"
 #include "media/cdm/ppapi/cdm_logging.h"
```
These include statements are Chromium specific. There will be a need to rebase the 'chromium-65' branch on the 'chromium-53' branch. See the following commit where they were modified to support WPE:

git show 109ca12dcb772fa05a663d165ae7be2d43f442b7

petegriffin committed:
BUILD.gn: add target_gen_dir to include_dirs

This allows us to #include opencdm_features.h
rather than having to specify the full path.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

ajutras-linaro commented:
Thank you Peter for the new patch.
